policyengine_uk_data/datasets/frs/childcare/takeup_rate.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from scipy.optimize import minimize
+from policyengine_uk import Microsimulation
+
+# 🎯 Calibration targets
+targets = {
+    "spending": {
+        "tfc": 0.6,
+        "extended": 2.5,
+        "targeted": 0.6,
+        "universal": 1.7,
+    },
+    "caseload": {
+        "tfc": 960,
+        "extended": 740,
+        "targeted": 130,
+        "universal": 490,
+    },
+}
+
+
+# 📦 Simulation runner
+def run_simulation(params, seed=42):
```

Comment:
```suggestion
def run_simulation(tfc_takeup_rate: float, seed=42):
```

Always add type hints, descriptive variable names, and a docstring with descriptions.

Reply:
added.